### PR TITLE
Change is_valid_identifier() in ustring to be similar to is_valid_identifier() in text_server.

### PR DESCRIPTION
**Pull request by magian1127**
Make it possible to use UTF in places like in the image below.

image

### CODE REVIEW

File: core/string/char_range.inc [hunk 1]
```diff
@@ -392,6 +392,7 @@ static CharRange xid_start[] = {
 	{ 0xab5c, 0xab69 },
 	{ 0xab70, 0xabe2 },
 	{ 0xac00, 0xd7a3 },
+	{ 0xb0a1, 0xf7fe },
```
Comment (bruvzg) [inline core/string/char_range.inc:395]: We do not use GB2321 encoding anywhere, all strings are UTF-32 or UTF-8. All ranges are UTF-32 ranges.

And for UTF-32 this is definitely wrong, this includes surrogates (`D800—DB7F` and `DC00—DFFF`) which should never be part of any string. And the rest should be already covered by other ranges.

**Review by bruvzg — CHANGES_REQUESTED** [review]
I'm not sure if it is a good idea to change `is_valid_identifier()`. Most likely there are other parts of the GDScript parser that will need to be changed (and if we decide to do it, it should use a proper implementation for `TextServer`).

File: core/string/char_range.inc (hunk 1 above)
Comment (magian1127) [reply core/string/char_range.inc:395]: I made a mistake... should use unicode 4e00-9fa5.
Actually, CharRange already contains.

**Comment by AThousandShips** [comment]
Many different places use this function and rely on it functioning as it does, there are a lot of places in code where this could cause errors

**Pull request closed without merging**